addons/mail/static/src/messaging/entity/entity/entity.js

```diff
@@ -0,0 +1,824 @@
+                Object.defineProperty(this, classAttributeName, {
+                    get: () => this.observable[classAttributeName],
+                });
+            }
+
+            this.__init(this);
+            this.__update(this, data);
+        }
+
+        /**
+         * Should only be called from static `create` method, to avoid identity
+         * crisis as proxified (and non-) class instances (from observable).
+         *
+         * @param {Object} [data]
+         */
+        constructor(data) {
+            Object.defineProperty(this, 'env', {
+                get: () => this.constructor.env,
+            });
+            this.localId = this._createInstanceLocalId(data);
+            this.constructor.__init(this);
+        }
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        /**
+         * @static
+         * @returns {mail.messaging.entity.Entity[]} all instances of this entity
+         */
+        static get all() {
+            const { state } = this.env.store;
+            return Object.values(state.entities).filter(entity => entity instanceof this);
+        }
+
+        /**
+         * Single way to create new entity instances in the store.
+         *
+         * @static
+         * @param {Object} data data object with initial data, including relations.
+         * @throws {Error} in case Entity class is a singleton and it creates
+         *   more than one instance.
+         * @returns {mail.messaging.entity.Entity} newly created entity
+         */
+        static create(data) {
+            const { state } = this.env.store;
+            if (this.isSingleton && this.all.length > 0) {
+                throw new Error(`Singleton Entity class "${this.name}" cannot create more than one instance.`);
+            }
+            const entity = new this(data);
+            state.entities[entity.localId] = entity;
+            // ensure observable version of entity is handled.
+            const proxifiedEntity = state.entities[entity.localId];
+            this.__update(proxifiedEntity, data);
+            return proxifiedEntity;
+        }
+
+        /**
+         * @static
+         */
+        static delete() {
+            for (const entity of this.all) {
+                entity.delete();
+            }
+            this.__delete(this);
+        }
+
+        /**
+         * @static
+         * @param {any} id
+         * @returns {mail.messaging.entity.Entity|undefined}
+         */
+        static fromId(id) {
+            const allEntities = this.all;
+            return allEntities.find(entity => entity.id === id);
+        }
+
+        /**
+         * @static
+         * @param {string|mail.messaging.entity.Entity|undefined} entityOrLocalId
+         * @returns {mail.messaging.entity.Entity|undefined}
+         */
+        static get(entityOrLocalId) {
```

Comment:
Explain the method. When is it ever useful to use `get` when we already have the entity to pass as parameter?